### ISSUE TITLE
"Unexpected token. Expected one of: ." at line 225

### ISSUE THREAD

**Issue by zzyyzz1992** [prisma/prisma2#266]
Here is my [schema.prisma](https://gist.github.com/zzyyzz1992/aa13d3742373be7645867720260a5e71),
It was base on prisma 1.34, introspected by `prisma2 init`.
I took hours to try my best to fix the grammer error.
But the vscode still tell me : "Unexpected token. Expected one of: ." at line 225
Same error when I execute `prisma2 generate`.

**Comment by pantharshit00** [comment]
Hey @zzyyzz1992 

Can you please share your Prisma 1.34 datamodel?

**Comment by zzyyzz1992** [comment]
@pantharshit00 Hi there, I finally figured out.
**I think prisma need to process these situation correctly:**
1)Enum value must not contain underscore(_)， in my case, when I remove all underscores, the problem page in vscode is correctly reporting problems in my schema, rather than tell me : "Unexpected token. Expected one of: ." 

**Besides,
VSCode reported those which I think shouldn't be problems:**
1)@@unique seems not supported, so I comment them temporally.
![image](https://user-images.githubusercontent.com/3894825/62043355-b7024900-b232-11e9-8af9-0fea4a52752b.png)
![image](https://user-images.githubusercontent.com/3894825/62043385-cb464600-b232-11e9-94e4-52af26afbb33.png)
2)After this, the vscode shows the schema has no problems, but when I execute `prisma2 generate`, error showed up on the terminal as I pasted below, which complains 
`Did not find a relation for those for model Relation_Employer and field createdDeliveryOrders'`
```
Error: process exited with a non-zero code: 101
thread 'main' panicked at 'Did not find a relation for those for model Relation_Employer and field createdDeliveryOrders', src/libcore/option.rs:1034:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:197
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:381
   6: rust_begin_unwind
             at src/libstd/panicking.rs:308
   7: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
   8: core::option::expect_failed
             at src/libcore/option.rs:1034
   9: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
  10: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter
  11: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
  12: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter
  13: prisma_models::datamodel_converter::DatamodelConverter::convert
  14: prisma::data_model_loader::load_data_model_components
  15: prisma::main
  16: std::rt::lang_start::{{closure}}
  17: std::panicking::try::do_call
             at src/libstd/rt.rs:49
             at src/libstd/panicking.rs:293
  18: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:87
  19: std::rt::lang_start_internal
             at src/libstd/panicking.rs:272
             at src/libstd/panic.rs:388
             at src/libstd/rt.rs:48
  20: main
  21: __libc_start_main
  22: _start
```
which I think maybe the relation name is duplicated, so I changed from
```
model Relation_Employer {
    ...
    createdOrderInfos  Order_Info[] @relation("creator")
    modifiedOrderInfos Order_Info[] @relation("modifier")
    createdDeliveryOrders  WH_DeliveryOrder[] @relation("creator")
    modifiedDeliveryOrders WH_DeliveryOrder[] @relation("modifier")

```

```
model WH_DeliveryOrder {
    ....
    creator   Relation_Employer                 @relation("creator")
    modifier  Relation_Employer?                @relation("modifier")
```
to :
```
model Relation_Employer {
    ...
    createdOrderInfos  Order_Info[] @relation("creator")
    modifiedOrderInfos Order_Info[] @relation("modifier")

    createdDeliveryOrders  WH_DeliveryOrder[] @relation("creatorOfDO")
    modifiedDeliveryOrders WH_DeliveryOrder[] @relation("modifierOfDO")
```

```
model WH_DeliveryOrder {
    ....
    creatorOfDO   Relation_Employer                 @relation("creatorOfDO")
    modifierOfDO  Relation_Employer?                @relation("modifierOfDO")
```
which fix the error.
**I consider this a bug, because I think it's normal when a `employer `being as creator of a `order`, and being a creator of a `delivery order`** at the same time.

Here is [the final schema file](https://gist.github.com/zzyyzz1992/b14bce7a6433f1e2e68635fb5b068854), which vscode says no problems anymore. But when I execute prisma2 generate, there're still errors, which I can not fix, please help me with this:
```
root@Dizzy-X-Pro:/mnt/c/Users/dizzy/Codes/tests/prisma2-test# prisma2 generate
Generating Photon JS Client to /mnt/c/Users/dizzy/Codes/tests/prisma2-test/prisma/node_modules/@generated/photon
Errors during Photon generation:
Return type of public method from exported class has or is using private name 'FindOneRelation_UnitArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneAccounting_PaymentArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBaseInfo_NormalGoodArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBaseInfo_TyrePatternSpecArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneOrder_ItemArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneOrder_InOrderItemArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneOrder_OutOrderItemArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_UnitArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneOrder_ItemArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneWH_TyreArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneOrder_ItemArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_CustomerAddressArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneOrder_InOrderInfoArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneOrder_OutOrderInfoArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBusiness_ApplicationArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_CustomerAddressArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneAccounting_PayAndGatheringArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_LogisticArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneOffice_UserTokenArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_UnitArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBaseInfo_PhotoArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_CustomerAddressArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_CustomerMoneyPeriodArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_CustomerQuotaArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_UnitArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBaseInfo_BrandArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBaseInfo_SpecArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBaseInfo_TyreInfoArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBaseInfo_FileArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneOrder_OutOrderInfoArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneWH_DeliveryOrderArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_UnitArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneRelation_UnitArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBaseInfo_TyrePatternSpecArgsRequired'.
Return type of public method from exported class has or is using private name 'FindOneBaseInfo_TyrePatternSpecArgsRequired'.
Done in 10.54s
```
**In Conclusion, I think there are following problems prisma2 needs to investigate further:
1) The Underscore Problem
2) @@unique not supported
3) Relation name duplicated in [Multiple-Reference Relationships](https://github.com/prisma/specs/tree/master/prisma-schema-language#multiple-reference-relationships,)**
Finally, the  `prisma generate`  still produce errors as I mentioned above,
I think I just need the final step to make prisma2 working, please help me with this, I'll be appreciated.

**Comment by zzyyzz1992** [comment]
I figure out that the [Multiple-Reference Relationships](https://github.com/prisma/specs/tree/master/prisma-schema-language#multiple-reference-relationships) are not necessarily making field name equals to __name__ in @relation(__name__) .
I think the spec is misleading, besides, a foreign key usually named like "FK_TABLENAME_xxx"

**Comment by zzyyzz1992** [comment]
@pantharshit00 
Due to multiple problems I reported in one issue, do I need to separate problems into multiple issues?

**Comment by pantharshit00** [comment]
Yes, please separate them out and close this one. Thanks for reporting :pray:

**Comment by zzyyzz1992** [comment]
@pantharshit00 closing issue, please check them out, thanks.
